### PR TITLE
Add basic design system page for new and edit pages

### PR DESCRIPTION
**Pull request by davidgisbey**
## Description

This is (finally 🎉 ) the first PR that starts to actually port stuff over to use the GOV.UK Design System. Going forward i'm going to go with the strategy of duplicating the new and edit page and slowly porting over the views one partial (or a couple if they're small) at a time so it can be reviewed fairly easily.

As i update a partial i will update any affected tests.

This PR deals with adding a `shell` for the new and edit edition pages. It only does the following

1. Adds a get layout method in the controller which renders the design system layout for the new, edit, create and update actions
2. Renders the `new` and `edit` template in place of the `new_legacy` and `edit_legacy` views when the user has the "Preview design system" permission
3. Adds a `new` and `edit` view which continues to render the legacy partials so all the tests keep passing
4. Adds a `shell` in the GOV.UK Design System (title, backlink, error summary etc.)

### New page

image

### Edit page

image

## Trello card
https://trello.com/c/YbYQICAe/684-move-edit-publication-page-to-the-govuk-design-system

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: app/views/admin/editions/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<% content_for :page_title, "Edit publication"%>
+<% content_for :back_link, (render "govuk_publishing_components/components/back_link", {
+    href: admin_edition_path(@edition)
+  })
+%>
+<% content_for :error_summary, render('shared/error_summary', object: @edition.becomes(Edition), class_name: @edition.class.name.demodulize.underscore.humanize.downcase) %>
```
Comment (kevindew) [inline app/views/admin/editions/edit.html.erb:6]: Corr this is a mouthful! Perhaps there's a future scope for a user_facing_name method on Editions. Not in this PR of course as consistent with legacy view.

File: app/views/admin/editions/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<% content_for :page_title, "Edit publication"%>
```
Comment (kevindew) [inline app/views/admin/editions/edit.html.erb:1]: Is this meant to use a variable instead of "publication" ?

**Review by kevindew — APPROVED** [review]
Amazing stuff @davidgisbey - really great to see this coming to life

File: app/views/admin/editions/edit.html.erb (hunk 1 above)
Comment (davidgisbey) [reply app/views/admin/editions/edit.html.erb:6]: I think this exists actually 🤔 

Let me find it and update it 👍

**Pull request merged**